### ISSUE TITLE
Quick download issues

### ISSUE THREAD

**Issue by ersi-dnd** [OtterBrowser/otter-browser#800]
1. I'd like the Download/Transfer feature's protocol to be exactly like Wget. For example, when I take a download link in sourceforge and I remove the final /download from the URL, leaving the file name as the end of the URL, then Wget interprets redirects correctly and finds the file that was meant to be downloaded.

In Otter's Quick Download, when I enter the same URL, Otter quickly downloads something with that filename, but it's not the same thing.

2. Another thing I'd like in the Quick Download box is the ability to enter lists. Right now, the user can enter something in Quick Download box, but the Quick Download does not do anything. It only connects to the URL when the user presses Enter with the focus on the box. This implies that the user should be able to enter more URL's one after the other. 

And when simultaneous download URLs can be entered this way, it further implies that the entered URLs should be available somehow for viewing and editing before launching the download by Enter. E.g., clicking in the Quick Download box when some URL is already there should expand the existing URL, or there could be a button next to the box "Expand contents". 

How about these ideas?

**Comment by lollox** [comment]
I don't think this is something a browser should offer by default... And probably implementing all these in QT could be a bit hard ...

Shouldn't be simpler if these features will be added thought an external module that integrates wget itself? (something like the old Spiderzilla extension in Firefox)

**Comment by ersi-dnd** [comment]
My apologies for having no clue how it could be implemented simpler. At least it's probably clear that when users have a link that is downloadable via the ordinary download dialogue, they expect it to be downloadable via Quick Download box too, so the Quick Download box should be ready for redirects and things like that.

**Comment by lollox** [comment]
I referred to the list management not to the first issue ;)

**Comment by Emdek** [comment]
@ersi-dnd, personally I don't think that our UI suggests that it allows to input multiple URL at once, it is a simple single line input.

But we definitely should try harder to extract file name.

**Comment by ersi-dnd** [comment]
I just tried Quick download.

1. Have a default download folder, such as Downloads
2. Take a Sourceforge link, e.g. https://downloads.sourceforge.net/manjarolinux/manjaro-xfce-16.10.3-stable-x86_64.iso
3. Put it into the Quick download slot
4. Press Enter
Result: It downloads something irrelevant into /tmp

Desired behaviour: Download into the preset default folder and operate like `wget` does with that link, i.e. understand redirects.
